shuup/front/apps/product_filter/views.py

```diff
@@ -0,0 +1,266 @@
+
+from shuup.admin.form_part import (
+    FormPart,
+    FormPartsViewMixin,
+    SaveFormPartsMixin,
+    TemplatedFormDef
+)
+from shuup.admin.utils.views import CreateOrUpdateView
+from shuup.admin.shop_provider import get_shop
+from shuup.admin.toolbar import PostActionButton, Toolbar
+from shuup.utils.form_group import FormGroup
+from shuup.front.utils.views import cache_product_things
+from shuup.front.template_helpers.product import is_visible
+from shuup.front.utils.sorts_and_filters import (
+    get_product_queryset, get_query_filters, post_filter_products,
+    ProductListForm, sort_products
+)
+from shuup.core.models import (
+    Attribute,
+    Product,
+    Category,
+    Product,
+)
+from .forms import (
+    BaseFilterSettingsForm,
+    CategoriesFilterSettingsForm,
+    AttributesFilterSettingsForm,
+    ProductFilterForm,
+)
+from .models import (
+    BasicFilterSettingsModel,
+    CategoriesFilterSettingsModel,
+    BasicAttributesFilterSettingsModel,
+    AttributesFilterSettingsModel,
+    BASIC_ATTRIBUTE_FIELDS,
+    LAYOUT_CHOICES,
+)
+
+
+class BaseFilterSettingsFormPart(FormPart):
+    priority = 1
+    name = 'base_settings'
+    form = BaseFilterSettingsForm
+
+    def get_form_defs(self):
+        yield TemplatedFormDef(
+            self.name,
+            self.form,
+            template_name='shuup/product_filter/admin/edit_base_settings_form.jinja',
+            required=True,
+            kwargs={
+                'instance': self.object
+            }
+        )
+
+    def form_valid(self, form):
+        self.object = form[self.name].save()
+        self.object.shop = get_shop(self.request)
+        self.object.save()
+        return self.object
+
+
+class CategoriesFilterSettingsFormPart(FormPart):
+    priority = 2
+    name = 'categories_settings'
+    form = CategoriesFilterSettingsForm
+
+    def get_form_defs(self):
+        try:
+            instance = CategoriesFilterSettingsModel.objects.filter(
+                shop=get_shop(self.request)
+            ).first()
+        except:
+            instance = CategoriesFilterSettingsModel()
+        yield TemplatedFormDef(
+            self.name,
+            self.form,
+            template_name='shuup/product_filter/admin/edit_categories_settings_form.jinja',
+            required=True,
+            kwargs={
+                'instance': instance,
+            }
+        )
+
+    def form_valid(self, form):
+        self.object = form[self.name].save()
+        self.object.shop = get_shop(self.request)
+        self.object.save()
+        return self.object
+
+
+class AttributesFilterSettingsFormPart(FormPart):
+    priority = 3
+    name = 'attributes_settings'
+    form = AttributesFilterSettingsForm
+
+    def get_form_defs(self):
+        try:
+            instance = AttributesFilterSettingsModel.objects.filter(
+                shop=get_shop(self.request)
+            ).first()
+        except AttributesFilterSettingsModel.DoesNotExist:
```

Comment:
This shouldn't be raised either. `instance` should be `None` when no attribute filters settings are found.